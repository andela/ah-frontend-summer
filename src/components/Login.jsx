import React from "react";
import PropTypes from 'prop-types';
import { Form, Grid, Segment, Message, Divider, Header } from "semantic-ui-react";
import classes from "../styles/login.module.css";
import '../styles/login.css';


const Login = props => {
    const {
        email,
        password,
        onDismissHandler,
        loginError,
        onSubmitHandler,
        onChangeHandler,
        loading,
        onBlurHandler,
        isEmailValid
    } = props;

    const loginErorMsg = (
        <Message
            negative
            content={loginError}
            onDismiss={onDismissHandler}
            style={{ width: '50%', margin: 'auto' }}
        />
    );
    const emailInvalidError = <div style={{ color: "red" }}>Please enter a valid email</div>;
    return (
        <div className={classes.LoginPage}>
            <Grid columns={2} stackable>
                <Grid.Column width={7}></Grid.Column>
                <Grid.Column
                    width={8}
                    padded="very"
                >
                    <Segment padded="very" color="green">

                        <Header as='h1' textAlign='center' padded="very">Author's Haven</Header>
                        <Header as='h6' style={{ color: "#008080", marginTop: "5px", textAlign: 'center', fontSize: "1em" }}>A home for the creative at heart</Header>

                        <br />
                        {loginError ? loginErorMsg : null}
                        <Form onSubmit={onSubmitHandler} className={classes.cardForm}>
                            <center>
                                <Form.Field>
                                    <input
                                        style={!isEmailValid ? { borderColor: "red", width: '50%' } : { width: '50%' }}
                                        type="text"
                                        placeholder="Email"
                                        name="email"
                                        value={email}
                                        onChange={onChangeHandler}
                                        onBlur={onBlurHandler}
                                    />
                                    {!isEmailValid ? emailInvalidError : null}
                                </Form.Field>
                                <Form.Field>
                                    <input
                                        type="password"
                                        placeholder="Password"
                                        name="password"
                                        value={password}
                                        onChange={onChangeHandler}
                                        style={{ width: '50%' }}
                                    />
                                </Form.Field>
                                <Form.Button type="submit" className="login" loading={loading}>Sign in</Form.Button>
                            </center>
                        </Form>
                        <Divider horizontal>Or</Divider>
                        <br />
                        <div className="spaceComponents">
                            <Header style={{ color: "#008080" }} textAlign='center' padded="very">Sign up</Header>
                        </div>

                    </Segment>
                </Grid.Column>
            </Grid>
        </div>
    );
};

Login.propTypes = {
    email: PropTypes.string.isRequired,
    password: PropTypes.string.isRequired,
    onDismissHandler: PropTypes.func.isRequired,
    loginError: PropTypes.string.isRequired,
    onSubmitHandler: PropTypes.func.isRequired,
    onChangeHandler: PropTypes.func.isRequired,
    loading: PropTypes.bool.isRequired,
    onBlurHandler: PropTypes.func.isRequired
};

export default Login;
