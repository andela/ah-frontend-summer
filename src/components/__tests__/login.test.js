import React from "react";
import { mount } from "enzyme";
import Login from "../Login";

describe("Login", () => {
    const onChangeHandlerMock = jest.fn();
    const onSubmitHandlerMock = jest.fn();
    const onBlurHandlerMock = jest.fn();
    const loginProps = {
        onChangeHandler: onChangeHandlerMock,
        loginError: "hahhjjke",
        onSubmitHandler: onSubmitHandlerMock,
        email: "",
        password: "",
        onDismissHandler: jest.fn(),
        onBlurHandler: onBlurHandlerMock,
        loading: false
    };
    const login = mount(<Login {...loginProps} />);

    it("renders login component correctly", () => {
        expect(login).toMatchSnapshot();
    })

    it("calls onChandeHandler when the email or password input is changed", () => {
        login.find('input[name="email"]').simulate("change");
        expect(onChangeHandlerMock).toHaveBeenCalled();
        login.find('input[name="password"]').simulate("change");
        expect(onChangeHandlerMock).toHaveBeenCalled();
    });

    it("renders Message component when loginError is true", () => {
        expect(login.find("Message").exists()).toBe(true);
    });

    it("calls onSubmitHandler when form is submited", () => {
        login.find("form").simulate("submit");
        expect(onSubmitHandlerMock).toHaveBeenCalled();
    });
});
