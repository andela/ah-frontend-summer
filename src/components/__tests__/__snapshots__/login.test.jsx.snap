// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders login component correctly 1`] = `
<Login
  email=""
  loading={false}
  loginError="whsdhfish"
  onChangeHandler={[MockFunction]}
  onDismissHandler={[MockFunction]}
  onSubmitHandler={[MockFunction]}
  password=""
>
  <div
    className="LoginPage"
  >
    <Grid
      columns={2}
      stackable={true}
    >
      <div
        className="ui stackable two column grid"
      >
        <GridColumn
          width={7}
        >
          <div
            className="seven wide column"
          />
        </GridColumn>
        <GridColumn
          padded="very"
          width={8}
        >
          <div
            className="eight wide column"
            padded="very"
          >
            <Segment
              color="green"
              padded="very"
            >
              <div
                className="ui green very padded segment"
              >
                <Header
                  as="h1"
                  padded="very"
                  textAlign="center"
                >
                  <h1
                    className="ui center aligned header"
                    padded="very"
                  >
                    Author's Haven
                  </h1>
                </Header>
                <Header
                  as="h6"
                  style={
                    Object {
                      "color": "#008080",
                      "fontSize": "1em",
                      "marginTop": "5px",
                      "textAlign": "center",
                    }
                  }
                >
                  <h6
                    className="ui header"
                    style={
                      Object {
                        "color": "#008080",
                        "fontSize": "1em",
                        "marginTop": "5px",
                        "textAlign": "center",
                      }
                    }
                  >
                    A home for the creative at heart
                  </h6>
                </Header>
                <br />
                <Message
                  content="whsdhfish"
                  negative={true}
                  onDismiss={[MockFunction]}
                  style={
                    Object {
                      "margin": "auto",
                      "width": "50%",
                    }
                  }
                >
                  <div
                    className="ui negative message"
                    style={
                      Object {
                        "margin": "auto",
                        "width": "50%",
                      }
                    }
                  >
                    <Icon
                      as="i"
                      name="close"
                      onClick={[Function]}
                    >
                      <i
                        aria-hidden="true"
                        className="close icon"
                        onClick={[Function]}
                      />
                    </Icon>
                    <MessageContent>
                      <div
                        className="content"
                      >
                        <p>
                          whsdhfish
                        </p>
                      </div>
                    </MessageContent>
                  </div>
                </Message>
                <Form
                  as="form"
                  className="cardForm"
                  onSubmit={[MockFunction]}
                >
                  <form
                    className="ui form cardForm"
                    onSubmit={[Function]}
                  >
                    <center>
                      <FormField>
                        <div
                          className="field"
                        >
                          <input
                            name="email"
                            onChange={[MockFunction]}
                            placeholder="Email"
                            style={
                              Object {
                                "borderColor": "red",
                                "width": "50%",
                              }
                            }
                            type="text"
                            value=""
                          />
                          <div
                            style={
                              Object {
                                "color": "red",
                              }
                            }
                          >
                            Please enter a valid email
                          </div>
                        </div>
                      </FormField>
                      <FormField>
                        <div
                          className="field"
                        >
                          <input
                            name="password"
                            onChange={[MockFunction]}
                            placeholder="Password"
                            style={
                              Object {
                                "width": "50%",
                              }
                            }
                            type="password"
                            value=""
                          />
                        </div>
                      </FormField>
                      <FormButton
                        as={[Function]}
                        className="login"
                        control={[Function]}
                        loading={false}
                        type="submit"
                      >
                        <FormField
                          className="login"
                          control={[Function]}
                          loading={false}
                          type="submit"
                        >
                          <div
                            className="field login"
                          >
                            <Button
                              as="button"
                              loading={false}
                              type="submit"
                            >
                              <button
                                className="ui button"
                                onClick={[Function]}
                                type="submit"
                              >
                                Sign in
                              </button>
                            </Button>
                          </div>
                        </FormField>
                      </FormButton>
                    </center>
                  </form>
                </Form>
                <Divider
                  horizontal={true}
                >
                  <div
                    className="ui horizontal divider"
                  >
                    Or
                  </div>
                </Divider>
                <br />
                <div
                  className="spaceComponents"
                >
                  <Header
                    padded="very"
                    style={
                      Object {
                        "color": "#008080",
                      }
                    }
                    textAlign="center"
                  >
                    <div
                      className="ui center aligned header"
                      padded="very"
                      style={
                        Object {
                          "color": "#008080",
                        }
                      }
                    >
                      Sign up
                    </div>
                  </Header>
                </div>
              </div>
            </Segment>
          </div>
        </GridColumn>
      </div>
    </Grid>
  </div>
</Login>
`;
